### ISSUE TITLE
Card doesn't get a height with 2.3.11

### ISSUE THREAD

**Issue by theolof** [CosmicMind/Material#601]
I used 2.2.0 before and just updated to 2.3.11.

The width of the Card is before, but the height is very small, as if it doesn't contain anything.

```
        cardView = Card()
        cardView.translatesAutoresizingMaskIntoConstraints = false
        cardView.contentView = detailView
        cardView.contentViewEdgeInsetsPreset = .square3
        self.view.addSubview(cardView)
        self.view.addConstraint(NSLayoutConstraint(item: cardView, attribute: .top, relatedBy: .equal, toItem: bar, attribute: .bottom, multiplier: 1.0, constant: DEFAULT_MARGIN * 2))
        self.view.addConstraint(NSLayoutConstraint(item: cardView, attribute: .bottom, relatedBy: .equal, toItem: self.view, attribute: .bottom, multiplier: 1.0, constant: -DEFAULT_MARGIN * 4))
        self.view.addConstraint(NSLayoutConstraint(item: cardView, attribute: .leading, relatedBy: .equal, toItem: self.view, attribute: .leading, multiplier: 1.0, constant: DEFAULT_MARGIN * 2))
        self.view.addConstraint(NSLayoutConstraint(item: cardView, attribute: .trailing, relatedBy: .equal, toItem: self.view, attribute: .trailing, multiplier: 1.0, constant: -DEFAULT_MARGIN * 2))
        self.view.addConstraint(NSLayoutConstraint(item: cardView, attribute: .centerX, relatedBy: .equal, toItem: self.view, attribute: .centerX, multiplier: 1.0, constant: 0))

```

where detailView is a view with TextFields, buttons, labels etc.

Has anything changed I should be aware of?

**Comment by theolof** [comment]
It works fine if `detailView` is a `UILabel()`. Then the card expands and shows the label.
In my case, the `detailView` is a `UIView()`.

**Comment by theolof** [comment]
I tried your Card sample now. It looks fine with the `contentView` as a `UILabel` as in the example.
I'm trying to modify your example to show a `UIView` instead as the `contentView`, but whenever I do that, it shows up empty. I even tried adding a `UILabel` to the `UIView` - still an empty `contentView`.

**Comment by danieldahan** [comment]
Hey, the `detailView` is the `contentView`. There has been a change to the math behind the card. It is now calculated without AutoLayout. This was done for performance reasons. In general though, the UIView that you place needs to have an `intrinsicContentSize` or a height given to it, otherwise, it doesn't know how to calculate the size. 

That said, the `contentView` property itself is a UIView, but not all views will be able to determine their content size dynamically. So you would need to set a height for instance. 

```swift
card.contentView = UIView()
card.contentView?.backgroundColor = Color.grey.base
card.contentView?.height = 100
```

When you set a UILabel as a subView of the `contentView` it is even further removed from making a  dynamic calculation. So you would need to manage the size of that content itself, then everything else will fall into place. 

If you have something you are trying to create, send me your sample code and I would be happy to help :)

**Comment by theolof** [comment]
Please tell me if I should send the file in another way.
I attached a ViewController here, it's from your Card sample.

I'm trying to add two TextFields to the contentView (via detailView). Ideally I don't want to set the height since I want the Card to be expanding to the contents of detailView so it looks good on different phones. Even when I do set the height on contentView, the TextFields don't show. Please help :-)

[ViewController.txt](https://github.com/CosmicMind/Material/files/599451/ViewController.txt)

**Comment by danieldahan** [comment]
I am going to try your sample today :)

**Comment by danieldahan** [comment]
So I tried your example, and without setting the height, it is not possible to make it work. So dynamic height is not built in. That said you inspired a new feature. I am going to make an additional layout tool that will make these situations simple to handle. I will followup in a couple days with a solution for you.

**Comment by theolof** [comment]
That would be awesome. I'm updating the whole codebase with the new Material but this is stopping me since a lot of my Cards have dynamic heights.

**Comment by danieldahan** [comment]
Wonderful, I am happy to help :)

My aim is to have a solution that you can test within a few days. So hopefully that wont block you too much.

**Comment by danieldahan** [comment]
Sorry for any delays. I have been a bit held up on some other tasks. I should be able to focus on this in the next couple days.

**Comment by danieldahan** [comment]
I am going to link this in with another issue and move this to a general `Layout` update feature request: https://github.com/CosmicMind/Material/issues/621. For now the issue is a limitation and not a bug and I believe I have a suggestion. I am going to think about layouts over the next couple days. Suggestions in the other issue are more than welcome :)
